src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,157 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private Node<T> prev;
+        private T element;
+        private Node<T> next;
+
+        private Node(Node<T> prev, T element, Node<T> next) {
+            this.prev = prev;
+            this.element = element;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> addNode = new Node<>(tail, value, null);
+        if (size == 0) {
+            head = tail = addNode;
+            size++;
+            return;
+        }
+        tail.next = addNode;
+        tail = addNode;
+        size++;
```

Comment:
I think it will be better to write
```suggestion
        if (size != 0) {
            tail.next = newNode;
        } else {
            head = newNode;
        }    
        tail = newNode;
        size++;
```